packages/report-server/src/types.ts

```diff
@@ -18,9 +18,19 @@ export interface FetchReportParams extends ParamsDictionary {
   reportCode: string;
 }
 
+export interface ReportConfig {
+  fetch: {
+    dataElements?: string[];
+    dataGroups?: string[];
+  };
+  transform: (string | Record<string, unknown>)[];
+}
+
 interface ReportsRequestBody {
   emailAddress: string;
   password: string;
+  testConfig?: ReportConfig;
+  testData?: Record<string, string | number>[];
```

Comment:
Added these two optional parameters, which allow to provide either a `testConfig` object (representing a report.config object) or `testData` representing an array or analytics or events. These can be really useful for testing when data isn't available, or for iteratively developing a transform config without needing to edit the database.

I'm thinking that ultimately it might be nice to break this out into a separate endpoint (perhaps `/report/:id/fetch` and `/report/:id/test`) but I still haven't made my mind up about it yet

Reply:
A separate endpoint would be great for keeping things clean and decoupled! As long as the actual behaviour can easily be invoked with test data passed in (to avoid duplication)

I guess your current implementation is still pretty clean, so we may as well leave as is for now

Reply:
Yeah I think both endpoints would use the same `ReportBuilder` class, just only one has the ability to add the test config / test data. 

The idea around splitting into separate endpoints is partially cos the decoupling would be nice, but also a security thing. Regular users should be able to fetch data, but only Admin users (or some other group) should be able to test different configs out

Reply:
Most definitely!